server/athenian/api/controllers/miners/github/release_load.py

```diff
@@ -374,481 +689,199 @@ def remove_ambigous_precomputed_releases(df: pd.DataFrame, repo_column: str) ->
+            if not regexp.endswith("$"):
+                regexp += "$"
+            # note: dict.setdefault() is not good here because re.compile() will be evaluated
+            try:
+                regexp = regexp_cache[regexp]
+            except KeyError:
+                regexp = regexp_cache[regexp] = re.compile(regexp)
+            matched = repo_branches[repo_branches[Branch.branch_name.key].str.match(regexp)]
+            if not matched.empty:
+                branches_matched[repo] = matched
+        return branches_matched
+
+    @sentry_span
+    @cached(
+        exptime=60 * 60,  # 1 hour
+        serialize=pickle.dumps,
+        deserialize=pickle.loads,
+        # commit_shas are already sorted
+        key=lambda commit_shas, time_from, time_to, **_: (",".join(commit_shas),
+                                                          time_from, time_to),
+        refresh_on_access=True,
+        cache=lambda self, **_: self._cache,
+    )
+    async def _fetch_commits(self,
+                             commit_shas: Sequence[str],
+                             time_from: datetime,
+                             time_to: datetime) -> pd.DataFrame:
+        if (min(time_to, datetime.now(timezone.utc)) - time_from) > timedelta(hours=6):
+            query = \
+                select([PushCommit]) \
+                .where(and_(PushCommit.sha.in_(commit_shas),
+                            PushCommit.committed_date.between(time_from, time_to),
+                            PushCommit.acc_id.in_(self._meta_ids))) \
+                .order_by(desc(PushCommit.committed_date))
+        else:
+            # Postgres planner sucks in this case and we have to be inventive.
+            # Important: do not merge these two queries together using a nested JOIN or IN.
+            # The planner will go crazy and you'll end up with the wrong order of the filters.
+            rows = await self._mdb.fetch_all(
+                select([NodeCommit.id])
+                .where(and_(NodeCommit.oid.in_any_values(commit_shas),
+                            NodeCommit.acc_id.in_(self._meta_ids),
+                            NodeCommit.committed_date.between(time_from, time_to))))
+            if not rows:
+                return pd.DataFrame(columns=[c.key for c in PushCommit.__table__.columns])
+            ids = [r[0] for r in rows]
+            assert len(ids) <= len(commit_shas), len(ids)
+            query = \
+                select([PushCommit]) \
+                .where(and_(PushCommit.node_id.in_(ids),
+                            PushCommit.acc_id.in_(self._meta_ids))) \
+                .order_by(desc(PushCommit.committed_date))
+        return await read_sql_query(query, self._mdb, PushCommit)
+
+
+# TODO: these have to be removed, these are here just for keeping backward-compatibility
+# without the need to re-write already all the places these functions are called
+load_releases = ReleaseLoader.load_releases
```

Comment:
Nothing's changed in the logic, you've simply wrapped them in the class, correct?

Reply:
yup, only wrapping 👍